admin/javascript/src/components/form-builder/index.js

```diff
@@ -0,0 +1,229 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as schemaActions from 'state/schema/actions';
+import SilverStripeComponent from 'silverstripe-component';
+import FormComponent from 'components/form/index';
+import TextField from 'components/text-field/index';
+import HiddenField from 'components/hidden-field/index';
+import GridField from 'components/grid-field/index';
+import fetch from 'isomorphic-fetch';
+
+import es6promise from 'es6-promise';
+es6promise.polyfill();
+
+// Using this to map field types to components until we implement dependency injection.
+const fakeInjector = {
+
+  /**
+   * Components registered with the fake DI container.
+   */
+  components: {
+    TextField,
+    GridField,
+    HiddenField,
+  },
+
+  /**
+   * Gets the component matching the passed component name.
+   * Used when a component type is provided bt the form schema.
+   *
+   * @param string componentName - The name of the component to get from the injector.
+   *
+   * @return object|null
+   */
+  getComponentByName(componentName) {
+    return this.components[componentName];
+  },
+
+  /**
+   * Default data type to component mappings.
+   * Used as a fallback when no component type is provided in the form schema.
+   *
+   * @param string dataType - The data type provided by the form schema.
+   *
+   * @return object|null
+   */
+  getComponentByDataType(dataType) {
+    switch (dataType) {
+      case 'String':
+        return this.components.TextField;
+      case 'Hidden':
+        return this.components.HiddenField;
+      case 'Text':
+        // Textarea field (not implemented)
```

Comment:
Since this is just a placeholder anyway, makes sense to add these as we create them? I assume dataType isn't going to eventually be a fixed list? (Will check form RFC now, maybe it makes it clearer).